utils/docker/Dockerfile.centos.7

```diff
@@ -60,7 +60,7 @@ gpgcheck=False\n" >> /etc/yum.repos.d/$repo:$branch:$build_number.repo;   \
 
 # Install OS updates and package.  Include basic tools and daos dependencies
 COPY ./utils/scripts/install-centos7.sh /tmp/install.sh
-RUN chmod +x /tmp/install.sh && dnf upgrade && /tmp/install.sh && dnf clean all && \
+RUN chmod +x /tmp/install.sh && /tmp/install.sh && dnf clean all && \
```

Comment:
This will mean that upgrades from REPOs won't be applied, is that intended or desirable?

Reply:
It is intended, and I believe it's desirable.  The REPOS will get the @PR version we request.

Reply:
Thinking about it I wonder if we want to keep this line as-is.  It won't do anything if the dockerfile is run in one go, however there might be several days between the previous RUN step and this one so it's possible both would end up installing packages.  As this RUN step will need to download the repo metadata anyway there's no advantage I can see to removing the upgrade here.

Reply:
The issue I'm trying to resolve, by moving the upgrade, is that custom repos added in (such as libfabric@PR-61, or mpifileutils-pkg@PR-23) were causing dnf upgrade to fail.

This could be seen in https://build.hpdd.intel.com/blue/organizations/jenkins/daos-stack%2Fdaos/detail/PR-9108/5/pipeline/191/ 
* (Search for the "docker build" in the Centos7.9 stage.)
  *   Hangs at 
  `"Upgrading        : centos-release-7-9.2009.1.el7.centos.x86_64           1/92The command '/bin/sh -c chmod +x /tmp/install.sh && dnf -y upgrade && /tmp/install.sh && dnf -y clean all &&     rm -f /tmp/install.sh' returned a non-zero code: 137"
`

Reply:
I've seen that bug and assumed it was something to do with the custom packages it was installing.

If "dnf upgrade" is failing the fix is not to avoid doing a dnf upgrade, we need to find out why it was failing and address the root cause.

Reply:
Would love to get a direct fix for this.

Would you take a look at https://daosio.atlassian.net/browse/DCO-10410?  

I'd appreciate any feedback you could give as to furthering this rabbit hole.  The issue has been hard to pin down.